### PR TITLE
In DbKnowledgeRepository, better handling of parent arg in _kp_dir

### PR DESCRIPTION
**Pull request by sryza**
Description of changeset:
This addresses issue #155, which was causing issues not to show up in knowledge posts when using a database repository.  The change fixes the `_kp_dir` method of `DbKnowledgeRepository` to look for the parent arg in the `ref` column of the database.

Test Plan:
Without this change, images didn't display in my knowledge posts.  With the change, they did.

Auto-reviewers: @NiharikaRay @matthewwardrop @earthmancash @danfrankj

### CODE REVIEW

**Comment by matthewwardrop** [comment]
Hey @sryza! Sorry for the radio silence again. Pushing toward some rapidly approaching deadlines.

I'll do a more thorough review of the patch after Friday, but in a quick run through I have a few thoughts.

The first one is just that you are using `os.path` rather than `posixpath`; which means that on Windows there may be some inconsistencies in the way files are stored in the database. Probably worth fixing before we merge this.

I also am a bit worried about some edge cases in the modified path lookup algorithm, but it may be nothing.  Just need some time to think through the possible cases.

Will give you some more thorough thoughts once I finish the race toward EOY deadlines.

Thanks again so much for your patch and for helping us to make the Knowledge Repo amazing.

**Comment by sryza** [comment]
Oops, didn't think posixpath had a relpath function, but it looks like it does.  Made that change.

File: knowledge_repo/repositories/dbrepository.py [hunk 1]
```diff
@@ -204,15 +204,15 @@ def _kp_read_ref(self, path, reference, revision=None):
...
+        ref_prefix = parent + '/' if parent else ''
         revision = revision or self._kp_get_revision(path, enforce_exists=True)
         refs = (self.session.query(self.PostRef.ref)
                             .filter(self.PostRef.path == path)
+                            .filter(self.PostRef.ref.like(ref_prefix + '%'))
                             .filter(self.PostRef.revision == revision)).all()
         for (ref,) in refs:
             if ref is not None:
-                yield ref
+                yield posix.relpath(ref, parent)
```
Comment (matthewwardrop) [inline knowledge_repo/repositories/dbrepository.py]: This should be `posixpath.relpath`, not `posix.relpath`.

**Comment by matthewwardrop** [comment]
Going to release a new version to fix some issues that we've been having with the knowledge_repo script, and so taking the liberty of merging this in now with my trivial fix applied.

Thanks again @sryza. We really appreciate your work! Looking forward to more patches!

**Comment by sryza** [comment]
Thanks @matthewwardrop !

**Pull request merged**